packages/legend-query/src/stores/QueryBuilderDateCapabilityHelper.ts

```diff
@@ -0,0 +1,588 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  type Enum,
+  type PureModel,
+  EnumValueExplicitReference,
+  EnumValueInstanceValue,
+  GenericType,
+  GenericTypeExplicitReference,
+  matchFunctionName,
+  PrimitiveInstanceValue,
+  PRIMITIVE_TYPE,
+  SimpleFunctionExpression,
+  TYPICAL_MULTIPLICITY_TYPE,
+} from '@finos/legend-graph';
+import {
+  guaranteeNonNullable,
+  UnsupportedOperationError,
+} from '@finos/legend-shared';
+import {
+  DAY_OF_WEEK,
+  DURATION_UNIT,
+  SUPPORTED_FUNCTIONS,
+} from '../QueryBuilder_Const';
+import { buildPrimitiveInstanceValue } from './QueryBuilderOperatorsHelper';
+import type { QueryBuilderState } from './QueryBuilderState';
+
+export type CustomDateInputOption = {
+  label: string;
+  value: string;
+};
+
+export type DateOption = {
+  label: string;
+  value: string;
+  durationRange?: string | undefined;
+  unit?: string | undefined;
+  adjustment?: string | undefined;
+  startDate?: string | undefined;
+};
+
+export const QUERY_BUILDER_RESERVED_DATETIME_OPERATOR_FUNCTION_TO_LABEL =
+  new Map([
+    [SUPPORTED_FUNCTIONS.IS_AFTER_DAY, ['>']],
+    [SUPPORTED_FUNCTIONS.IS_BEFORE_DAY, ['<']],
+    [SUPPORTED_FUNCTIONS.IS_ON_DAY, ['is', 'is not']],
+    [SUPPORTED_FUNCTIONS.IS_ON_OR_AFTER_DAY, ['>=']],
+    [SUPPORTED_FUNCTIONS.IS_ON_OR_BEFORE_DAY, ['<=']],
+  ]);
+
+export enum QUERY_BUILDER_DATE_OPTONS {
+  TODAY = 'Today',
+  NOW = 'Now',
+  YESTERDAY = 'Yesterday',
+  ONE_YEAR_AGO = 'One Year Ago',
+  ONE_MONTH_AGO = 'One Month Ago',
+  ONE_WEEK_AGO = 'One Week Ago',
+  CUSTOM_DATE = 'Custom Date',
+  ABSOLUTE_DATE = 'Absolute Date',
+  PREVIOUS_DAY_OF_WEEK = 'Previous ... of Week',
+  START_DAY_OF_DATE = 'Start of...',
+}
+
+export enum QUERY_BUILDER_DATE_DURATION_UNIT_LABELS {
```

Comment:
just our convention for `enum`s, don't do plural, --> `QUERY_BUILDER_DATE_DURATION_UNIT_LABEL`